File: codigos/1 - estrutura sequencial/1018 - cedulas/Main.dart
import 'dart:io';

main(){
  int valor = int.parse(stdin.readLineSync()!);
  stdout.writeln(valor);

  List<int> cedulas = [100, 50, 20, 10, 5, 2, 1];

  for (int i = 0; i < cedulas.length; i++){
    int cedula = cedulas[i];
    int qntd = (valor/cedula).floor();
    valor -= qntd*cedula;
    print("$qntd nota(s) de R\$ $cedula,00");
  }
}

File: codigos/1 - estrutura sequencial/1018 - cedulas/Main.java
import java.io.IOException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        int valor = new Scanner(System.in).nextInt();
        System.out.println(valor);
        int[] cedulas = {100, 50, 20, 10, 5, 2, 1};
        for (int cedula:cedulas) {
            int qntd_cedulas_temp;
            if (valor >= cedula){
                qntd_cedulas_temp = valor/cedula;
                System.out.println(qntd_cedulas_temp + " nota(s) de R$ " + cedula + ",00");
                valor -= qntd_cedulas_temp * cedula;
            } else {
                qntd_cedulas_temp = 0;
                System.out.println(qntd_cedulas_temp + " nota(s) de R$ " + cedula + ",00");
            }
        }
    }
}